lib/application.js

```diff
@@ -60,11 +61,30 @@ module.exports = class Application extends Emitter {
 
   listen(...args) {
     debug('listen');
-    const server = http.createServer(this.callback());
+    const server = this.server = http.createServer(this.callback());
     return server.listen(...args);
```

Comment:
I think we don't need to add a close method because it rarely used, and we can close http server like this:

```js
const app = new Koa();
const server = app.listen();
server.close();
```

Reply:
Yep. You are right! I will close this PR later